repository/src/main/java/org/apache/atlas/repository/audit/ESBasedAuditRepository.java

```diff
@@ -243,6 +248,35 @@ private EntityAuditSearchResult getResultFromResponse(String responseString) thr
                 eventKey = event.getEntityId() + ":" + event.getTimestamp();
             }
 
+            Map<String, Object> detail = event.getDetail();
+            if (detail != null && detail.containsKey("attributes")) {
+                Map<String, Object> attributes = (Map<String, Object>) detail.get("attributes");
+                List<AtlasEntityHeader> linkedEntityList = new ArrayList<>();
+
+                for (Map.Entry<String, Object> entry: attributes.entrySet()) {
+                    if (guidKeys.contains(entry.getKey()) && entry.getValue() instanceof List) {
+                        List<String> guids = (List<String>) entry.getValue();
+
+                        if (guids != null && !guids.isEmpty()){
```

Comment:
guids.isEmpty already checks for null and empty. Take a look at the documentation of the method and update this.